ext/include/opentelemetry/ext/zpages/zpages.h

```diff
@@ -10,48 +12,34 @@
 #include "opentelemetry/sdk/trace/tracer_provider.h"
 #include "opentelemetry/nostd/shared_ptr.h"
 
- /*
-  * Wrapper class that allows users to use zPages by calling this class constructor. Currently only
-  * has TraceZ, but other types of zPages can be added in the future
-  */
-class zPages {
- public:
-  zPages()
-    : tracez_processor_(std::make_shared<opentelemetry::ext::zpages::TracezSpanProcessor>()),
-      tracez_provider_(opentelemetry::nostd::shared_ptr<opentelemetry::trace::TracerProvider>(
-        new opentelemetry::sdk::trace::TracerProvider(tracez_processor_))) {
-
-    // Set the global trace provider for a user to use, which is connected to our span processor
-    opentelemetry::trace::Provider::SetTracerProvider(tracez_provider_);
 
-    tracez_server_thread_ = std::thread(&zPages::RunTracezServer, this);
-    tracez_server_thread_.detach();
-
-    // Ensure zPages has time to setup, so the program doesn't crash
-    std::this_thread::sleep_for(setup_time_);
+class zPages {
+public:
+  static void Initialize(){
+    static zPages instance;
```

Comment:
static zPages* instance = new zPages;

Reply:
as per our discussion offline I've left this as is for the champions to review.